### PR TITLE
packfile/scanner: reset zlib reader instead of new one

### PR DESCRIPTION
**Pull request by ajnavarro**

### CODE REVIEW

File: plumbing/format/packfile/scanner.go [hunk 1]
```diff
@@ -275,19 +276,28 @@ func (s *Scanner) NextObject(w io.Writer) (written int64, crc32 uint32, err erro
 // ReadRegularObject reads and write a non-deltified object
 // from it zlib stream in an object entry in the packfile.
 func (s *Scanner) copyObject(w io.Writer) (int64, error) {
```
Comment (alcortesm) [inline plumbing/format/packfile/scanner.go]: The comment on this function starts with a wrong function name.

File: plumbing/format/packfile/scanner.go [hunk 2]
```diff
@@ -275,19 +276,28 @@ func (s *Scanner) NextObject(w io.Writer) (written int64, crc32 uint32, err erro
...
-	zr, err := zlib.NewReader(s.r)
-	if err != nil {
-		return -1, fmt.Errorf("zlib reading error: %s", err)
+	var err error
+	if s.zr == nil {
+		zr, err := zlib.NewReader(s.r)
+		if err != nil {
+			return -1, fmt.Errorf("zlib initialization error: %s", err)
```
Comment (alcortesm) [inline plumbing/format/packfile/scanner.go]: Why are we returning -1, this is not C!

`Scanner.NextObject` first return argument is not used anywhere, it is safe to return 0 here instead.

File: plumbing/format/packfile/scanner.go (hunk 1 above)
Comment (ajnavarro) [inline plumbing/format/packfile/scanner.go]: I think this change is not related to this PR, Could fix this naming issue into another PR?

File: plumbing/format/packfile/scanner.go (hunk 2 above)
Comment (ajnavarro) [inline plumbing/format/packfile/scanner.go]: I was just replying the previous behavior. I can change it.

File: plumbing/format/packfile/scanner.go (hunk 1 above)
Comment (smola) [inline plumbing/format/packfile/scanner.go]: Yes, please. Avoid unrelated changes here.

Comment (alcortesm) [inline plumbing/format/packfile/scanner.go]: Sure, change it on another PR.
